src/App.jsx

```diff
@@ -0,0 +1,285 @@
+        };
+        setToasts([...toasts, newToast]);
+      } else if (flag === "incomplete") {
+        const { data, error } = await supabase
+          .from("ReactTodo")
+          .select()
+          .ilike("name", `%${search}%`)
+          .is("completed_on", null)
+          .order("id", { ascending: false });
+
+        if (error === null) {
+          setTodos([]);
+          setTodos(data);
+        } else {
+          setFlag(prevFlag);
+          setFlagChange(false);
+        }
+
+        newToast = {
+          id: uuidv4(),
+          type: error ? "error" : "success",
+          message: error ? error.message : "Incompleted Data fetched",
+        };
+
+        setToasts([...toasts, newToast]);
+      } else {
+        const { data, error } = await supabase
+          .from("ReactTodo")
+          .select()
+          .ilike("name", `%${search}%`)
+          .order("id", { ascending: false })
+          .not("completed_on", "is", null);
+        if (error === null) {
+          setTodos([]);
+          setTodos(data);
+        } else {
+          setFlag(prevFlag);
+          setFlagChange(false);
+        }
+
+        newToast = {
+          id: uuidv4(),
+          type: error ? "error" : "success",
+          message: error ? error.message : "Completed Data fetched",
+        };
+        setToasts([...toasts, newToast]);
+      }
+
+      setShowBigSpinner(false);
+    };
+    fetchData();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [flag, search]);
+
+  //Toast useeffect
+  useEffect(() => {
+    if (toasts.length > 0) {
+      setTimeout(() => {
+        let allToast = [...toasts];
+        allToast.shift();
+        setToasts(allToast);
+      }, 500);
+    }
+  }, [toasts]);
+
+  const handleRemoveTodo = async (id) => {
+    let newToast;
+    // eslint-disable-next-line
+    const { data, error } = await supabase
+      .from("ReactTodo")
+      .delete()
+      .match({ id: id });
+
+    if (error === null) {
+      setDataCount(dataCount - 1);
+      removeCompleteFromIncomplete(id);
+      if (dataCount === 0) {
+        setSearchShow(false);
+        setFlag("all");
+      }
+    }
+    newToast = {
+      id: uuidv4(),
+      type: error ? "error" : "success",
+      message: error ? error.message : "Task Deleted",
+    };
+    setToasts([...toasts, newToast]);
+  };
+
+  return (
+    <div className="App">
+      {splash && (
+        <div>
+          <div className="container">
+            {" "}
+            <div
+              className={classNames({
+                "progress-bar": progress,
+                "fullprogress-bar": !progress,
+              })}
+            ></div>
+          </div>
+          <div className="splash faddingOut  ">
+            <div className="splashLogo">
+              <div className="splashIconLogo">
+                <Icon src="SplashLeaf" />
+              </div>
+              <div>
+                <Icon src="SplashTitle" />
```

Comment:
Can you implement this in BEM structure?

Reply:
implemented BEM structure